src/AuthorizationServiceInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authorization;
+
+/**
+ * Interface for Authorization service
+ */
+interface AuthorizationServiceInterface
+{
+    /**
+     * Check whether the provided user can perform an action on a resource.
+     *
+     * @param \Authorization\IdentityInterface $user The user to check permissions for.
+     * @param string $action The action/operation being performed.
+     * @param mixed $resource The resource being operated on.
+     * @return bool
+     */
+    public function can(IdentityInterface $user, $action, $resource);
```

Comment:
Not sure if the service should typehint here.

Reply:
I'd relax this parameter and allow anything to be passed.

Reply:
Done!